src/agent.rs

```diff
@@ -97,11 +104,36 @@ impl AgentCollection {
         let mut cashs: Vec<_> = self.agents.iter().map(|a| a.cash).collect();
         cashs.sort_by(|a, b| a.partial_cmp(b).unwrap());
         cashs[cashs.len() / 2]
-    }
+    }// Making decision based on 
 
-    pub fn assign_states(&mut self) {
-        // Every agent updates their beliefs based on other agents' interests,
-        // overall profits from a market and other random factors.
+    pub fn update_behaviour(&mut self) {
+        // Update trading behaviour based on how much the agent won/lost.
+        // Gambler's fallacy + noise
+        // Variables to update: fundamentalism_ratio(?), order_probability
         todo!()
     }
+
+    pub fn update_market_interest(&mut self, market: &GenoaMarket) {
+        // Every agent updates their beliefs based on other agents' preferences
+        // and their own interests. At every time step, the interest for a market
+        // is updated based on performance (overall profits from a market), news
+        // and random noise.
+        let mut rng = rand::thread_rng();
+        let m_id = market.id() as usize;
+        let mut rand_idx: usize;
+        let mut market_noise: f32;
+
+        for idx in 0..self.agents.len() {
+            rand_idx = rng.gen_range(0..self.agents.len());
+            market_noise = rng.gen_range(0.0..1.0);
+
+            self.agents[idx].interest_prob_vector[m_id] = ((1.
+                - self.agents[idx].fundamentalism_ratio)            
+                * self.agents[rand_idx].interest_prob_vector[m_id]  // Making decision based on herd behaviour.
+                + self.agents[idx].fundamentalism_ratio
+                * (market.get_markup() + market.get_news())         // Making decision based on fundamentals.
+                + market_noise)                                 // Adding noise to decision.
+                .tanh();                                            // Keeping the value between 0 and 1.
```

Comment:
I assume you did not run `cargo fmt`, but this will most likely be messed up. Better to split it up into variables.

Reply:
I did run `cargo fmt`. That's why the `((1.` is placed after the equal.